### ISSUE TITLE
YAMNet sidecar: TFLite runtime (option 3 from #491) to cut 1.56GB to ~250MB, fidelity-gated

### ISSUE THREAD

**Issue by sydlexius** [sydlexius/canticle#503]
[mode: plan] 
[model: sonnet] 
[effort: medium] 

## Description

Follow-up to #491. That issue evaluated three ways off `tensorflow-hub` and took **option 1** (vendor the model, load it directly with `tf.saved_model.load`), which cleared the setuptools CVE and cut the image from 2.24GB to 1.56GB. Options 2 (ONNX) and 3 (TFLite) were explicitly left on the table. This is option 3.

Current state on `main` after #491 / #497:

```
canticle-yamnet:pre491           2.24GB   (before #491)
canticle-yamnet:local            1.56GB   (after #491)
ghcr.io/doxazo-net/canticle       162MB
```

`deploy/yamnet-detector/requirements.in` now pins `tensorflow-cpu==2.18.1` with `tensorflow-hub` gone and the setuptools ceiling lifted. The lock is 59 packages and still contains `keras==3.15.0` and `tensorboard==2.18.0` -- a training API and a visualization dashboard in an inference-only container. Those arrive as `tensorflow-cpu`'s own dependencies, so they leave only when it does.

So the remaining question is narrow: we are still carrying a ~600MB general-purpose training runtime to run one frozen model whose weights are roughly 4MB. YAMNet has an official TFLite export, and `tflite-runtime` is roughly 50MB. Realistic landing zone is ~250MB.

## Why this is more than housekeeping

- **It decides whether an all-in-one image is viable.** At 1.56GB a bundled variant is unaffordable; at ~250MB it lands near 400MB, which is unremarkable. That measurement gates the AIO work.
- **CVE surface.** `make scan` gates on HIGH+ CVEs via grype. #491 was itself triggered by a setuptools CVE arriving through this dependency chain. `tensorflow-cpu` plus `keras` plus `tensorboard` is a large recurring surface carried for no inference reason, and this is the second CVE-driven issue in that chain.
- **arm64.** `tflite-runtime` has real arm64 wheels (it targets edge devices). Full TF on arm64 has always been the painful path, so this turns multi-arch from "probably not" into roughly free.

## The risk is framing, not the model

TFLite YAMNet is a conversion of the same weights, not a different model. The bar is **indistinguishable**, not "better" -- if the port comes out meaningfully different, that is a bug, not a result. Expecting an improvement is how a divergent port gets accepted as a win.

The real risk is windowing. The vendored SavedModel takes a variable-length waveform and frames internally; TFLite has a fixed ~0.975s input frame and the framing must be hand-rolled. Specifically: **a framing bug hides in `mean` and shows up in `max`.** Averaging washes out a framing error; a peak does not. `app.py` says so itself:

```
- "mean" feeds Canticle's music gate (summed mean of the instrumental classes ...)
- "max"  feeds Canticle's vocal gate. A sung track has a brief, strong singing ...
```

A brief, strong peak is exactly what bad framing destroys, and `max` feeds the vocal gate. So a framing bug would silently reclassify sung tracks as instrumental while `mean` looked healthy.

The contract is favorable: `POST /classify` returns `{"mean": {...}, "max": {...}}` over AudioSet class names, and the three gates live Go-side in `internal/detector`. The port only has to reproduce the score vectors. If they match, the tuned thresholds and `detector_version` comparability carry over untouched.

Use the float32 TFLite export, not int8 quantized, so drift stays near noise.

## Fidelity test

Differential, no ground-truth labels required: run both containers on different ports, POST the same files to each, diff the JSON. No canticle involvement, read-only, zero provider requests. Only disagreements need inspection. Both images already exist on the deployment host, so the harness is a script, not infrastructure.

A stronger end-to-end check is free: the DB holds ~11,500 rows of stamped telemetry from the current model (it stamps on both instrumental and non-instrumental verdicts), so the baseline is already recorded. Diff raw score dicts first -- it fails closer to the cause -- then validate the chain against stored telemetry.

## Acceptance Criteria

- [ ] `tensorflow-cpu`, `keras`, and `tensorboard` are gone from the lock
- [ ] The float32 TFLite YAMNet export is baked in at build time (no runtime network), preserving #491's vendored-model posture
- [ ] Image size materially reduced (target ~250MB, from 1.56GB)
- [ ] Differential test shows `mean` and `max` indistinguishable from the current model on a representative sample
- [ ] `max` divergence specifically is checked, as the framing-sensitive output feeding the vocal gate
- [ ] The three Go-side gates and their thresholds are unchanged
- [ ] `make scan` HIGH+ CVE surface is not regressed
- [ ] Hash-pinned requirements and digest-pinned base image discipline preserved

## Additional Context

- Follows #491 (closed; took option 1, deferred options 2/3). Related: #498 (sidecar has no published image or CI build -- this changes what that CI builds).
- Gates #504 (all-in-one image variant).
- If fidelity cannot be matched, the honest outcome is to keep `tensorflow-cpu` and close this. The size win is not worth moving a tuned gate.
